frontend/src/customHooks/useFetch.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+import { useState, useEffect } from "react";
+
+export function useFetch<T>(url: string) {
+  const [data, setData] = useState<T | null>(null);
+  const [isPending, setIsPending] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsPending(true);
+      try {
+        console.log("Fetching data from:", process.env.NEXT_PUBLIC_BASE_URL + url);
```

Comment:
The URL of the backend will be different than the NextJS app. Our backend is separate. Just a nit about the variable name, we should be able to set it either way.